test/jdk/java/lang/StringBuilder/HugeCapacity.java

```diff
@@ -75,7 +78,9 @@ private static void testUtf16() {
 
     private static void testHugeInitialString() {
         try {
-            String str = "Z".repeat(Integer.MAX_VALUE - 8);
+            // Maximum array length supported by the JVM implementation
+            int maxArrayLength = ArraysSupport.SOFT_MAX_ARRAY_LENGTH;
+            String str = "Z".repeat(maxArrayLength);
```

Comment:
The comment is a bit misleading. I'm not sure any comment is warranted, as you had observed, there's a big description next to SOFT_MAX_ARRAY_LENGTH. I'm also not sure a new local variable maxArrayLength adds anything.

Reply:
> The comment is a bit misleading. I'm not sure any comment is warranted, as you had observed, there's a big description next to SOFT_MAX_ARRAY_LENGTH. I'm also not sure a new local variable maxArrayLength adds anything.

Thanks, I have removed the comment and variable for these two tests.

I guess I felt SOFT_MAX_ARRAY_LENGTH looked a bit lonely here, given that it was just used as an argument and had no constant or comment like in most other instances. But I agree that the comment and the variable does not add much value here, the name of the test makes it clear that this is about capacity and the documentation on SOFT_MAX_ARRAY_LENGTH is just a click away.